numpy/ma/core.py

```diff
@@ -1032,12 +1032,7 @@ def __call__(self, a, b, *args, **kwargs):
         if m is not nomask and m.any():
             # any errors, just abort; impossible to guarantee masked values
             try:
-                np.copyto(result, 0, casting='unsafe', where=m)
-                # avoid using "*" since this may be overlaid
-                masked_da = umath.multiply(m, da)
-                # only add back if it can be cast safely
-                if np.can_cast(masked_da.dtype, result.dtype, casting='safe'):
-                    result += masked_da
```

Comment:
Not sure what the deal with that block was, I could only guess it did not used to be copyto and whatever was there before did not support broadcasting without trickery. Or maybe I should make it safe casting with the 0?